cmd/provision/provision.go

```diff
@@ -321,129 +330,35 @@ func (p *provision) verify(config *server.Config, verbosef shared.FormatFn) erro
-
-		config.Analytics.FluentdEndpoint = fmt.Sprintf(fluentdInternalFormat, p.Org, p.Env, p.Namespace)
-		config.Analytics.TLS.CAFile = defaultApigeeCAFile
-		config.Analytics.TLS.CertFile = defaultApigeeCertFile
-		config.Analytics.TLS.KeyFile = defaultApigeeKeyFile
-	}
-
-	if p.IsOPDK {
-		config.Analytics.LegacyEndpoint = true
-	}
-
-	return config
-}
+func (p *provision) verify(config *server.Config, verbosef shared.FormatFn) error {
 
-func (p *provision) printConfig(config *server.Config, printf shared.FormatFn, verifyErrors error) error {
-	// encode config
-	var yamlBuffer bytes.Buffer
-	yamlEncoder := yaml.NewEncoder(&yamlBuffer)
-	yamlEncoder.SetIndent(2)
-	err := yamlEncoder.Encode(config)
+	client, err := p.createAuthorizedClient(config)
 	if err != nil {
 		return err
 	}
-	configYAML := yamlBuffer.String()
-
-	data := map[string]string{"config.yaml": configYAML}
-	configCRD := server.ConfigMapCRD{
-		APIVersion: "v1",
-		Kind:       "ConfigMap",
-		Metadata: server.Metadata{
-			Name:      "apigee-remote-service-envoy",
-			Namespace: p.Namespace,
-		},
-		Data: data,
-	}
 
-	yamlBuffer.Reset()
-	yamlEncoder = yaml.NewEncoder(&yamlBuffer)
-	yamlEncoder.SetIndent(2)
-	err = yamlEncoder.Encode(configCRD)
-	if err != nil {
-		return err
+	var verifyErrors error
+	if p.IsLegacySaaS || p.IsOPDK {
+		verbosef("verifying internal proxy...")
+		verifyErrors = p.verifyInternalProxy(client, verbosef)
 	}
 
-	// secret for IsGCPManaged
-	if p.IsGCPManaged {
-		privateKeyBytes := pem.EncodeToMemory(&pem.Block{Type: server.PEMKeyType,
-			Bytes: x509.MarshalPKCS1PrivateKey(config.Tenant.PrivateKey)})
-
-		// create CRD for secret
-		jwksBytes, err := json.Marshal(config.Tenant.JWKS)
-		if err != nil {
-			return err
-		}
-
-		props := map[string]string{server.SecretPropsKIDKey: config.Tenant.PrivateKeyID}
-		propsBuf := new(bytes.Buffer)
-		if err := server.WriteProperties(propsBuf, props); err != nil {
-			return err
-		}
-
-		secretData := map[string]string{
-			server.SecretJKWSKey:    base64.StdEncoding.EncodeToString(jwksBytes),
-			server.SecretPrivateKey: base64.StdEncoding.EncodeToString(privateKeyBytes),
-			server.SecretPropsKey:   base64.StdEncoding.EncodeToString(propsBuf.Bytes()),
-		}
-
-		secretCRD := server.SecretCRD{
-			APIVersion: "v1",
-			Kind:       "Secret",
-			Type:       "Opaque",
-			Metadata: server.Metadata{
-				Name:      fmt.Sprintf(policySecretNameFormat, p.Org, p.Env),
-				Namespace: p.Namespace,
-			},
-			Data: secretData,
-		}
+	verbosef("verifying remote-service proxy...")
+	verifyErrors = multierr.Combine(verifyErrors, p.verifyRemoteServiceProxy(client, verbosef))
 
-		err = yamlEncoder.Encode(secretCRD)
-		if err != nil {
-			return err
+	if p.IsGCPManaged {
+		if version, err := p.checkRuntimeVersion(config, client, verbosef); err != nil {
+			err = errors.Wrapf(err, "Unable to get the runtime version")
+			verifyErrors = multierr.Combine(verifyErrors, err)
+		} else if version >= "1.3.0" {
```

Comment:
Since we're checking for the err here anyway, would it make sense to also call encodeUDCAEndpoint() if there's an error here - instead of also calling it from the timeout block?

Reply:
If we call it here, we will have to change it back if any subsequent verification succeeds.